collector/filesystem_common.go

```diff
@@ -126,30 +126,37 @@ func (c *filesystemCollector) Update(ch chan<- prometheus.Metric) (err error) {
 	if err != nil {
 		return err
 	}
+	// Make sure we expose a metric once, even if there are multiple mounts
+	seen := map[filesystemStats]bool{}
```

Comment:
While I believe the change to filesystemsStats was worthwhile anyway, it's actually still possible that we export multiple metrics for the same label set. Unless it's impossible in all os implementations to create structs with  different metric values but same label values, we should maybe afterall create a key only based on the label values.